### PR TITLE
[TR] Merge 1 4 into master

### PR DESCRIPTION
**Pull request by jjanvier**
TODO:
 - [x] merge https://github.com/akeneo/pim-community-dev/pull/3716 on 1.4 first
 - [x] try to bump Doctrine libs to stable versions
 - [ ] change the doc for Doctrine bump
 - [x] merge 1.4 on master on remove Benoits commits related to cascade detach
 - [x] retest import of 5k products if ensure memory leak problem is fixed on associations

### CODE REVIEW

File: src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php [hunk 1]
```diff
@@ -43,13 +47,15 @@ class ProductAssociationProcessor extends AbstractProcessor
...
+     * @param ObjectDetacherInterface               $detacher       detacher to remove it from UOW when skip
      */
     public function __construct(
         StandardArrayConverterInterface $arrayConverter,
         IdentifiableObjectRepositoryInterface $repository,
         ObjectUpdaterInterface $updater,
         ValidatorInterface $validator,
-        ProductFilterInterface $productAssocFilter
+        ProductFilterInterface $productAssocFilter,
+        ObjectDetacherInterface $detacher = null
```
Comment (jjanvier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php]: clean it

File: build.xml [hunk 2]
```diff
@@ -362,13 +360,7 @@
...
             <arg value="--no-update" />
             <arg value="doctrine/mongodb-odm-bundle" />
-            <arg value="v3.0.0-BETA6@dev" />
-        </exec>
-        <exec executable="${basedir}/composer.phar" failonerror="true">
-            <arg value="require" />
-            <arg value="--no-update" />
-            <arg value="doctrine/mongodb-odm" />
-            <arg value="v1.0.0-beta12@dev" />
+            <arg value="v3.0.1" />
```
Comment (skeleton) [inline build.xml]: :+1:

File: src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php [hunk 3]
```diff
@@ -205,4 +224,18 @@ protected function validateProductAssociations(ProductInterface $product)
...
+    /**
+     * Detaches the product from the unit of work is the responsibility of the writer but in this case we
+     * want ensure that an updated and invalid product will not be used in the association processor.
+     * Also we don't want to keep skipped products in memory
+     *
+     * @param ProductInterface $product
+     */
+    protected function detachProduct(ProductInterface $product)
+    {
+        if (null !== $this->detacher) {
```
Comment (skeleton) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php]: $this->detacher should not be null on master

**Comment by juliensnz** [comment]
GTM

**Comment by skeleton** [comment]
GTM

**Pull request merged**